src/helpers/isSupportedUriExtension.ts

```diff
@@ -0,0 +1,30 @@
+import { endsWith } from 'lodash';
+import parse from 'url-parse';
+
+// Defines whether a given uri corresponds to a set of possible file extensions.
+// It enforces the concept that uris must use https, that the evaluated url from
+// url-parse matches what has been provided and the url ends with a supported extension.
+export default function isSupportedUriExtension(
+  uri: string,
+  extensions: readonly string[]
+): boolean {
+  if (typeof uri !== 'string' || !Array.isArray(extensions)) {
+    return false;
+  }
+  try {
+    const { href, pathname, protocol } = parse(uri || '');
+    const supported = extensions.reduce(
+      (maybeSupported: boolean, ext: string): boolean => {
+        return (
+          maybeSupported ||
+          (typeof ext === 'string' &&
+            endsWith(pathname.toLowerCase(), ext.toLowerCase()))
```

Comment:
this is actually in a syntax of new JS
```suggestion
            pathname.toLowerCase().endsWith(ext.toLowerCase())
```

Reply:
I saw a similar operation was introduced recently for comparing URLs which used this approach, so I didn't want to regress that behaviour. But I can change this back.